core/domain/obj_services_test.py

```diff
@@ -63,3 +64,43 @@ def test_object_js_filenames(self):
                 self.assertIn(
                     'oppia.directive(\'%s\'' % normalized_directive_name,
                     template, msg='(%s)' % obj_type)
+
+
+class ObjectDefaultValuesUnitTests(test_utils.GenericTestBase):
+    """Test that the default value of objects recorded in
+    extensions/interactions/object_defaults.json correspond to
+    the defined default values in objects.py for all objects that
+    are used in rules."""
+
+    def test_all_rule_input_fields_have_default_values(self):
+        """Checks that all rule input fields have a default value, and this
+        is provided in get_default_values()."""
+
+        interactions = interaction_registry.Registry.get_all_interactions()
+        object_default_vals = obj_services.get_default_object_values()
+
+        for interaction in interactions:
+            for rule_name in interaction.rules.keys():
+                param_list = interaction.get_rule_param_list(rule_name)
+
+                for (_, param_obj_type) in param_list:
+                    param_name = param_obj_type.__name__
```

Comment:
Should param_name be param_type or param_type_name? Otherwise there's potential for confusion with the name of the parameter instance.